contacts/main.js

```diff
@@ -0,0 +1,563 @@
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Annette',
+    lastName: 'Carlson',
+    image: 'images/profile-pics/pic3.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Ervin',
+    lastName: 'Watkins',
+    image: 'images/profile-pics/pic4.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Philip',
+    lastName: 'Pope',
+    image: 'images/profile-pics/pic5.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Viorel',
+    lastName: 'Popescu',
+    image: 'images/profile-pics/pic6.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Elon',
+    lastName: 'Musk',
+    image: 'images/profile-pics/pic7.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  let retunrArray = [];
+  array.forEach((item) => {
+    let person = new Person(item);
+    retunrArray.push(person);
+  })
+  return retunrArray;
+}
+
+function saveDataToLocalStorage() {
+  localStorage.setItem('contacts', JSON.stringify(contactsList));
+}
+
+function getContactsFromLocalStorage() {
+  return JSON.parse(localStorage.getItem('contacts'));
+}
+
+function getContactsAccordingly() {
+  let array = getContactsFromLocalStorage();
+  // if we have nothing in the local storage we get the initial list
+  if (array === null || array.length === 0)
+    array = getInitialListOfContacts()
+  return array;
+}
+
+/////////////////////////////////////////////////////////////////////////
+
+function getNumberOfNonZeroLikes() {
+  let nonZeros = 0;
+  contactsList.forEach((person) => {
+    if (person.likes !== 0)
+      nonZeros += 1;
+  })
+  return nonZeros;
+}
+
+function refreshFavsList() {
+  // calculate the new list 
+  let favsMap = calculateFavsList();
+  // clear
+  clearFavsListHTML();
+  // rerender the new list if there is at least one nonzero elem
+  if (getNumberOfNonZeroLikes() !== 0)
+    rerenderFavsListHTML(favsMap);
+}
+
+function calculateFavsList() {
+  // returns a map
+  let maap = {};
+  // we deep copy it
+  let arr = [...contactsList];
+  // we sort it by likes
+  sortArrayByLikes(arr);
+  // we create the map of arrays
+  maap = createMapOfFavs(arr);
+  // sort each of them
+  Object.keys(maap).forEach((key) => {
+    sortArrayAlphabetically(maap[key]);
+  })
+  return maap;
+}
+
+function sortArrayByLikes(arrayOfContacts) {
+  // arrayOfContacts - array of objects person
+  arrayOfContacts.sort((p1, p2) => {
+    return p2.likes - p1.likes;
+  })
+}
+
+function createMapOfFavs(arrayOfContacts) {
+  let myMap = {};
+  arrayOfContacts.forEach((person) => {
+    let maximNoOfLikes = person.likes;
+    if (typeof myMap[maximNoOfLikes] === 'undefined') {
+      // we create a new one
+      myMap[maximNoOfLikes] = [];
+      myMap[maximNoOfLikes].push(person);
```

Comment:
myMap[maximNoOfLikes].push(person); is done both on the if and on the else branches. You can just make 1 push and move it outside the if clause